### ISSUE TITLE
[Bug]: Sandbox mode can use editable-installed source outside sandbox

### ISSUE THREAD

**Issue by aliengineeringbyte** [ilyuka/MLReproMutate#1]
### MLReproMutate version

0.1.1 

### Installation method

PyPI

### Mutation operator

data-split

### Python version

3.12

### Operating system

Kali Linux

### Command

```shell
mlrepromutate run . \
  --command "python -m pytest -q" \
  --operator data-split \
  --candidate-index 1 \
  --execution-mode sandbox
```

### Expected behavior

Sandbox mode should use the project code from sandbox copy.

The tests should run on mutated code, even when original project is installed with pip install -e ..

In this case mutation should be KILLED.

### Observed behavior

Sandbox mode can import package from original project folder instead of sandbox copy.

Because of this, mutation is not tested and result is SURVIVED.

If I add PYTHONPATH=src, then same mutation is correctly KILLED.

in-place mode also gives correct KILLED result.

### Relevant output or traceback

```text
Without PYTHONPATH:

SURVIVED
Summary: 1 mutations, 0 killed, 1 survived.

With sandbox src first:

KILLED
Summary: 1 mutations, 1 killed, 0 survived.
```

### Additional context

Hello sir,

My project is using src/ layout and pytest.

I have this code:

train_test_split(
    dataset.features,
    dataset.target,
    test_size=test_size,
    random_state=seed,
    stratify=dataset.target,
)

MLReproMutate finds this mutation:

Disable train/test split stratification for train_test_split.

I also have one test which checks that stratify=dataset.target is passed.

If I run sandbox mode with:

env PYTHONPATH=src python -m pytest -q

then mutation is correctly KILLED.

Also in-place mode works correctly.

It looks like editable install is taking priority over sandbox source path.

Maybe sandbox mode should put sandbox src path first in Python import path.

Thank you sir.

### Checklist

- [x] I have removed secrets and sensitive information from the report.
- [x] I have checked that this issue is not already reported.

**Comment by ilyuka** [comment]
Hi @aliengineeringbyte ,

Thanks for the detailed report. I was able to reproduce this independently with an editable-installed src/-layout project.
The root cause was that sandbox validation could resolve the package from the original project tree instead of the mutated sandbox copy, which could produce a false SURVIVED result.

I’ve added a regression test and a fix in #2. The original reproduction now reports KILLED in sandbox mode as expected, and the full test suite is passing.
